### ISSUE TITLE
Grafana 2 compatibility?

### ISSUE THREAD

**Issue by mikemccabe** [Kratos-ISE/grafana-kairosdb-datasource-plugin#8]
Is this plugin compatible with grafana 2?

**Comment by rdettai** [comment]
Sadly not. We don't know when we will have time to get to it. We will keep
you informed!

**Comment by br0th3r** [comment]
I just have uploaded this code to support Grafana 2 with KairosDB as a datasource. It is an adapted version from grafana-kairosdb-datasource-plugin mixed with OpenTSDB plugin for Grafana 2. :-)

**Comment by br0th3r** [comment]
Grafana 2 - KairosDB Datasource Plugin - https://github.com/br0th3r/grafana2-kairosdb-datasource-plugin

**Comment by rdettai** [comment]
Nice! 

Weird choice to license something that comes from Apache v2 (whole Grafana is Apache v2) into GPL... I believe it is your right (not an expert though), but it is not very good for us. This forbids us from using your contribution in our integrated solution that uses Grafana... I believe somebody will have to re-code it with a proper license !

**Comment by br0th3r** [comment]
Don't panic!, nobody has to recode anything. Is it OK now? :-)

Tomorrow we will upload several more changes, specially support for NULL points and other stuff we have repaired.

**Comment by rdettai** [comment]
As Tony the Tiger would say: This is GRRRRREEEAAAAT! ;-)

We will definitively test this as soon as you have finished. I believed Torkelo said somewhere that he wanted to maintain a version of grafana that would be fully client side. Do you know if this new plugin format is compatible with that?

**Comment by br0th3r** [comment]
Yes, totally. The plugins at the moment are sending their queries to Grafana 2 in the *usual format* for KairosDB. So Grafana 2 is just a *proxy* to KairosDB for this kind of queries.

Here you have an example of the query format:
```json
{"metrics":[{"name":"metric1","aggregators":[{"name":"avg","align_sampling":true,"sampling":{"value":"1"
,"unit":"minutes"}}]}],"cache_time":0,"start_absolute":1428544495222,"end_absolute":1428594702407}
```

and the answer:
```json
{"queries":[{"sample_size":3,"results":[{"name":"metric1","group_by":[{"name":"type","type":"number"}],"tags"
:{"lolo":["metric1"]},"values":[[1428573694970,22]]},{"name":"metric1","group_by":[{"name":"type","type":"text"
}],"tags":{"metric1":["metric1"]},"values":[[1428573694969,"None"],[1428573694971,"None"]]}]}]}
```

The url was:
```http://localhost:3000/api/datasources/proxy/4/api/v1/datapoints/query```

So I don't see why not to have Grafana 2 fully client side. :+1:

**Comment by masaori335** [comment]
Hi, guys

As @torkelo [commented](https://github.com/Kratos-ISE/grafana-kairosdb-datasource-plugin/issues/4#issuecomment-92608704) in #4, KairosDB Plugin ( compatible with Grafana 2.x ) is in a branch of grafana repo.
See : https://github.com/grafana/grafana/tree/kariosdb
As far as I tried it works fine.
